lib/graphql/analysis/ast/query_complexity.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+module GraphQL
+  module Analysis
+    # Calculate the complexity of a query, using {Field#complexity} values.
+    module AST
+      class QueryComplexity < Analyzer
+        # State for the query complexity calcuation:
+        # - `complexities_on_type` holds complexity scores for each type in an IRep node
+        def initialize(query)
+          super
+          @complexities_on_type = [TypeComplexity.new]
+          @skip_stack = [false]
+          @in_fragment_def = false
```

Comment:
It looks like some of these IVars were moved to the parent class, is that right?